src/pygeoprocessing/routing/routing.pyx

```diff
@@ -138,6 +138,18 @@ cdef struct FlowPixelType:
     int last_flow_dir
     double value
 
+cdef struct DecayingValue:
+    double decayed_value  # The value, which will be progressively updated as it decays
+    double min_value  # The minimum value before the Decaying Value should be ignored.
+

+cdef struct WeightedFlowPixelType:
+    int xi
+    int yi

```

Comment:
The other `struct` definitions use `unsigned int` for pixel location, any reason not to do that here as well for consistency?

Reply:
Nope!  In fact they really should be `unsigned int`s for consistency with the way indexes are used elsewhere in the function.  Patched in d645b70a